client/trino-cli/src/main/java/io/trino/cli/Trino.java

```diff
@@ -34,12 +44,42 @@ public static void main(String[] args)
 
     public static CommandLine createCommandLine(Object command)
     {
-        return new CommandLine(command)
+        CommandLine commandLine = new CommandLine(command)
                 .registerConverter(ClientResourceEstimate.class, ClientResourceEstimate::new)
                 .registerConverter(ClientSessionProperty.class, ClientSessionProperty::new)
                 .registerConverter(ClientExtraCredential.class, ClientExtraCredential::new)
                 .registerConverter(HostAndPort.class, HostAndPort::fromString)
                 .registerConverter(Duration.class, Duration::valueOf);
+
+        getConfigFile().ifPresent(file -> ValidatingPropertiesDefaultProvider.attach(commandLine, file));
+        return commandLine;
+    }
+
+    private static Optional<File> getConfigFile()
+    {
+        return getConfigSearchPaths()
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(Paths::get)
+                .filter(Files::exists)
+                .findFirst()
+                .map(Path::toFile);
+    }
+
+    private static Optional<String> resolveConfigPath(String root, String file)
```

Comment:
this should be below `getConfigSearchPaths`

Reply:
Sure. I'll push it before merge.